### PR TITLE
Update to Rust 2018

### PR DESCRIPTION
**Pull request by thomaslienbacher**
Changes:
* Updated code
* Updated examples in the documentation
* Added my name to the Cargo.toml files

I didn't update the version number, because I don't know if this would equal a major or a minor update.
Addresses this issue: #7 

This is my first pull request, so please give me some hints on how to make them better.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 /target
 **/*.rs.bk
+.idea
```
Comment (cjhowe-us) [inline .gitignore]:
```suggestion
```
I used to do this too, before I learned about setting up a [global `.gitignore`](https://help.github.com/articles/ignoring-files/#create-a-global-gitignore) for these kinds of rules. I don't think this change is necessary.

File: named_type_derive/src/lib.rs [hunk 2]
```diff
@@ -3,12 +3,10 @@
...
 
-extern crate proc_macro;
-extern crate syn;
-#[macro_use]
-extern crate quote;
-extern crate named_type;
+extern crate proc_macro; // must use extern crate for proc macros
+                         // see here: https://users.rust-lang.org/t/how-to-use-proc-macro-on-rust-2018/20833/2
 
+use quote::*;
```
Comment (cjhowe-us) [inline named_type_derive/src/lib.rs]:
```suggestion
use quote::{Tokens, quote};
```

I'd prefer to explicitly import only what we use in the namespace. Note that there's a `quote::Tokens` in this file which would no longer need the prefix.

**Comment by cjhowe-us** [comment]
I left some review comments. I think this is going to be a patch update because it's mostly documentation changes. [Technically it doesn't matter](https://semver.org/#spec-item-4) according to semver for versions less than 1.0 though. Thanks for contributing!

**Review by cjhowe-us — CHANGES_REQUESTED** [review]

**Comment by thomaslienbacher** [comment]
I have applied your suggestions and updated the version number, by incrementing the patch number.
I think this branch is ready to be squashed and merged.

**Review by cjhowe-us — APPROVED** [review]

**Comment by cjhowe-us** [comment]
Merged. Thank you!

**Pull request closed without merging**